src/integer_mod_q/mat_zq/get.rs

```diff
@@ -0,0 +1,372 @@
+use crate::utils::coordinate::evaluate_coordinates;
+use crate::{error::MathError, integer_mod_q::Zq};
+use flint_sys::fmpz::{fmpz, fmpz_set};
+use flint_sys::fmpz_mod_mat::fmpz_mod_mat_entry;
+use std::fmt::Display;
+
+impl MatZq {
+    /// Returns the modulus of the matrix as a [`Modulus`].
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    ///
+    /// let matrix = MatZq::new(5, 10, 7).unwrap();
+    /// let entry = matrix.get_mod();
+    /// ```
+    pub fn get_mod(&self) -> Modulus {
+        let mut out = Z::default();
+        unsafe { fmpz_set(&mut out.value, &self.matrix.mod_[0]) };
+
+        Modulus::try_from_z(&out).expect("The matrix modulus is not a valid modulus.")
+    }
+}
+
+impl GetNumRows for MatZq {
+    /// Returns the number of rows of the matrix as a [`i64`].
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use math::traits::GetNumRows;
+    ///
+    /// let matrix = MatZq::new(5, 6, 7).unwrap();
+    /// let rows = matrix.get_num_rows();
+    /// ```
+    fn get_num_rows(&self) -> i64 {
+        self.matrix.mat[0].r
+    }
+}
+
+impl GetNumColumns for MatZq {
+    /// Returns the number of columns of the matrix as a [`i64`].
+    ///
+    /// # Example
+    /// ```
+    /// use math::integer_mod_q::MatZq;
+    /// use math::traits::GetNumColumns;
+    ///
+    /// let matrix = MatZq::new(5, 6, 7).unwrap();
+    /// let rows = matrix.get_num_columns();
+    /// ```
+    fn get_num_columns(&self) -> i64 {
+        self.matrix.mat[0].c
+    }
+}
+
+impl GetEntry<Z> for MatZq {
+    /// Outputs the [`Zq`] value of a specific matrix entry.
+    ///
+    /// Parameters:
+    /// - `row`: specifies the row in which the entry is located
+    /// - `column`: specifies the column in which the entry is located
+    ///
+    /// Returns the [`Zq`] value of the matrix at the position of the given
+    /// row and column or an error, if the number of rows or columns is
+    /// greater than the matrix or negative.
+    ///
+    /// # Example
+    /// ```rust
+    /// use math::integer_mod_q::MatZq;
+    /// use crate::math::traits::GetEntry;
+    /// use math::integer::Z;
+    ///
+    /// let matrix = MatZq::new(5, 10, 7).unwrap();
+    /// let entry: Z = matrix.get_entry(0, 1).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of rows or columns is greater than the matrix or negative.
+    fn get_entry(
+        &self,
+        row: impl TryInto<i64> + Display + Copy,
+        column: impl TryInto<i64> + Display + Copy,
+    ) -> Result<Z, MathError> {
+        let (row_i64, column_i64) = evaluate_coordinates(self, row, column)?;
+
+        let mut copy = fmpz::default();
+        let entry = unsafe { fmpz_mod_mat_entry(&self.matrix, row_i64, column_i64) };
+        unsafe { fmpz_set(&mut copy, entry) };
+
+        Ok(Z { value: copy })
```

Comment:
The less we expose the fmpz values the better. Less considerations regarding memory leaks
```suggestion
        let mut out = Z::default();
        let entry = unsafe { fmpz_mod_mat_entry(&self.matrix, row_i64, column_i64) };
        unsafe { fmpz_set(&mut out.value, entry) };

        Ok(out)
```